core/backend/src/ChangedElementsDb.ts

```diff
@@ -131,13 +127,49 @@ export class ChangedElementsDb implements IDisposable {
       options.wantParents,
       options.wantPropertyChecksums,
       options.rulesetDir,
-      options.tempDir
+      options.tempDir,
+    );
+    if (status !== DbResult.BE_SQLITE_OK)
+      throw new IModelError(status, "Failed to process changesets");
+    return status;
+  }
+
+  /** Processes a range of changesets and adds it to the changed elements cache
+   * This call will close the IModelDb object as it is required for processing and applying changesets
+   * @param requestContext The client request context
+   * @param briefcase iModel briefcase to use
+   * @param options options for processing
+   */
+  public async processChangesetsAndRoll(requestContext: AuthorizedClientRequestContext, briefcase: IModelDb, options: ProcessChangesetOptions): Promise<DbResult> {
+    requestContext.enter();
+    const changeSummaryContext = new ChangeSummaryExtractContext(briefcase);
+    const changesets = await ChangeSummaryManager.downloadChangeSets(requestContext, changeSummaryContext, options.startChangesetId, options.endChangesetId);
+    requestContext.enter();
+    const tokens = ChangedElementsDb.buildChangeSetTokens(changesets, BriefcaseManager.getChangeSetsPath(briefcase.iModelId));
+    // ChangeSets need to be processed from newest to oldest
+    tokens.reverse();
+    // Close briefcase before doing processing and rolling briefcase
+    const dbFilename = briefcase.pathName;
+    const dbGuid = briefcase.getGuid();
+    briefcase.close();
+    // Process changesets
+    const status: DbResult = this.nativeDb.processChangesetsAndRoll(
```

Comment:
what does "process changeset and roll" mean?

Reply:
Is just the version compare processing that does the following:
1. Obtain related property paths from older Db state (db file is in old state)
2. Applies changeset to move to new state ("rolls" the db forward)
3. Process it and finds changed elements

This is just to save us from cloning db files and applying changesets more than once (e.g. in addon code for processing and in iModelJs side for agent workflow)

Reply:
this must all have to be very internal and only used internal to the change summary agent, right (e.g. there had better be no local changes)? 

For V2 checkpoints it won't be possible, but then you'll be able to open both checkpoints directly.

Reply:
Yes, this is internal for our version comparison processing in an agent, not an active application's backend. Had a conversation with Bill G regarding V2 checkpoints, and yes, I'd have to have both checkpoints but it should be fairly performant. I'll keep conversing with Bill so that I may react when necessary.